service/history/workflow/mutable_state.go

```diff
@@ -362,6 +364,8 @@ type (
 		)
 		UpdateActivity(*persistencespb.ActivityInfo) error
 		UpdateActivityWithTimerHeartbeat(*persistencespb.ActivityInfo, time.Time) error
+		UpdateActivityWithCallback(*persistencespb.ActivityInfo, UpdateActivityCallback)
```

Comment:
Do you really need this method?
I saw that you use it in the api implementation, but you already have the `*persistencespb.ActivityInfo`, so why not pass that into the already existing `UpdateActivity`?

Reply:
Why is that a concern?

Reply:
note: one more refactoring PR to remove old UpdateActivity

Reply:
https://temporalio.atlassian.net/browse/OSS-3579